lib/board.rb

```diff
@@ -43,4 +46,32 @@ def empty_position?(input)
     position != "X" && position != "O"
   end
 
+  def won?
+    win_for("X") || win_for("O")
+  end
+
+  private
+
+  def win_for(mark)
+    winning_line(columns, mark) || winning_line(rows, mark) || winning_line(diagonals, mark)
+  end
+
+  def winning_line(lines, mark)
+    lines.map {|line| line.all? {|position|
```

Comment:
what happens if you change `map` to `any?`? Would you then be able to get rid of the `.include? true`?